merlin/main.py

```diff
@@ -897,6 +915,86 @@ def generate_worker_touching_parsers(subparsers: ArgumentParser) -> None:
         help="regex match for specific workers to stop",
     )
 
+    # merlin manager
+    manager: ArgumentParser = subparsers.add_parser(
+        "manager",
+        help="Watchdog application to manage workers",
+        description="A daemon process that helps to restart and communicate with workers while running.",
+        formatter_class=ArgumentDefaultsHelpFormatter,
+    )
+    manager.set_defaults(func=process_manager)
+
+    manager_commands: ArgumentParser = manager.add_subparsers(dest="command")
+    manager_run = manager_commands.add_parser(
+        "run",
+        help="Run the daemon process",
+        description="Run manager",
+        formatter_class=ArgumentDefaultsHelpFormatter,
+    )
+    manager_run.add_argument(
+        "-qf",
+        "--query_frequency",
+        action="store",
+        type=int,
+        default=60,
+        help="The frequency at which workers will be queried for response.",
+    )
+    manager_run.add_argument(
+        "-qt",
+        "--query_timeout",
+        action="store",
+        type=float,
+        default=0.5,
+        help="The timeout for the query response that are sent to workers.",
+    )
+    manager_run.add_argument(
+        "-wt",
+        "--worker_timeout",
+        action="store",
+        type=int,
+        default=180,
+        help="The sum total(query_frequency*tries) time before an attempt is made to restart worker.",
+    )
+    manager_run.set_defaults(func=process_manager)
+    manager_start = manager_commands.add_parser(
+        "start",
+        help="Start the daemon process",
+        description="Start manager",
+        formatter_class=ArgumentDefaultsHelpFormatter,
+    )
+    manager_start.add_argument(
+        "-qf",
+        "--query_frequency",
+        action="store",
+        type=int,
+        default=60,
+        help="The frequency at which workers will be queried for response.",
+    )
+    manager_start.add_argument(
+        "-qt",
+        "--query_timeout",
+        action="store",
+        type=float,
+        default=0.5,
+        help="The timeout for the query response that are sent to workers.",
+    )
+    manager_start.add_argument(
+        "-wt",
+        "--worker_timeout",
+        action="store",
+        type=int,
+        default=180,
+        help="The sum total(query_frequency*tries) time before an attempt is made to restart worker.",
+    )
```

Comment:
Thought about this more and decided it's better to have these options specifically for run and start like you have already. It would still be nice to reduce duplicate code though. I'll be making a PR soon with my suggested changes for this and other things I mentioned in my review.

Reply:
@bgunnar5 Include a nested function to reduce redundancy in PR